apps/backend/src/prisma-client.tsx

```diff
@@ -40,22 +43,33 @@ function getNeonPrismaClient(connectionString: string) {
     neonPrismaClient = new PrismaClient({ adapter });
     prismaClientsStore.neon.set(connectionString, neonPrismaClient);
   }
-
   return neonPrismaClient;
 }
 
 function getSchemaFromConnectionString(connectionString: string) {
   return (new URL(connectionString)).searchParams.get('schema') ?? "public";
 }
 
+async function resolveNeonConnectionString(entry: string): Promise<string> {
+  if (!isUuid(entry)) {
+    return entry;
+  }
+  const store = await stackServerApp.getDataVaultStore('neon-connection-strings');
+  const secret = getEnvVariable('STACK_SERVER_SECRET');
```

Comment:
I think it's actually better if we don't use STACK_SERVER_SECRET and instead just something like "no client side encryption" as a secret value (the secret doesn't actually need to be secret, it's only to hide it from Stack Auth, but we are Stack Auth so hiding it from ourselves doesn't make that much sense). thought being that if something ever happens to the encryption algorithm used by the store.getValue function that leaks the secret, we still want STACK_SERVER_SECRET to remain safe